### PR TITLE
Checkout fallback regex tweaks

### PR DESCRIPTION
**Pull request by mikejolley**
Closes #15161

### CODE REVIEW

File: assets/js/frontend/checkout.js [hunk 1]
```diff
@@ -426,14 +428,21 @@ jQuery( function( $ ) {
...
 
-							if ( null === valid_json ) {
+							if ( null === maybe_valid_json ) {
 								console.log( 'Unable to fix malformed JSON' );
 							} else {
-								console.log( 'Fixed malformed JSON. Original:' );
-								console.log( raw_response );
-								raw_response = valid_json[0];
+								// Check for valid JSON
+								data = $.parseJSON( maybe_valid_json[0] );
```
Comment (bor0) [inline assets/js/frontend/checkout.js]: Shouldn't this be wrapped in try/catch?

Comment (mikejolley) [inline assets/js/frontend/checkout.js]: A nested one you mean? Hmm. Maybe this needs a function handler.

Comment (coderkevin) [inline assets/js/frontend/checkout.js]: Yeah, I think I'd just make a function that validates JSON, catches exception and returns true/false. Then you could use it both places.

File: assets/js/frontend/checkout.js [hunk 2]
```diff
@@ -168,6 +168,19 @@ jQuery( function( $ ) {
 		reset_update_checkout_timer: function() {
 			clearTimeout( wc_checkout_form.updateTimer );
 		},
+		is_valid_json: function( raw_json ) {
+			try {
+				var json = $.parseJSON( raw_json );
+
+				if ( json && 'object' === typeof json ) {
```
Comment (bor0) [inline assets/js/frontend/checkout.js]: Can simplify to `return json && 'object' === typeof json`

**Review by bor0 — APPROVED** [review]

**Pull request merged**